recipes/tiny-dnn/all/conanfile.py

```diff
@@ -41,58 +34,69 @@ def _min_compilers_version(self):
             "gcc": "5",
             "clang": "3.4",
             "apple-clang": "10",
-            "Visual Studio": "14"
+            "Visual Studio": "14",
         }
 
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
     def requirements(self):
         self.requires("cereal/1.3.1")
         self.requires("stb/cci.20210713")
         if self.options.with_tbb:
             self.requires("onetbb/2020.3")
 
+    def package_id(self):
+        self.info.clear()
+
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._min_cppstd)
+            check_min_cppstd(self, self._min_cppstd)
 
         compiler = str(self.settings.compiler)
-        version = tools.Version(self.settings.compiler.version)
+        version = Version(self.settings.compiler.version)
         if compiler in self._min_compilers_version and version < self._min_compilers_version[compiler]:
-            raise ConanInvalidConfiguration(
-                "{} requires a compiler that supports at least C++{}".format(
-                    self.name, self._min_cppstd,
-                )
-            )
-
-    def package_id(self):
-        self.info.header_only()
+            raise ConanInvalidConfiguration(f"{self.name} requires a compiler that supports at least C++{self._min_cppstd}")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        # TODO: if you move this recipe to CMakeDeps, be aware that tiny-dnn
+        #       relies on CMake variables which are not defined in CMakeDeps, only
+        #       in cmake_find_package. So patch it before.
```

Comment:
?? This comment is from the old generator, is it a leftover and has it been fixed, or has the TODO not been applied?

Reply:
Uh, yes. It's from the original recipe, but it was not clear to me which variables the comment was referring to, so I left the comment intact. CMakeDeps appeared to work for me without issues.